File: Demo/AndroidFullStackParameter/app/src/main/java/com/example/eyeofcloud/androidfullstackparameter/LoadingActivity.java
package com.example.eyeofcloud.androidfullstackparameter;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.eyeofcloud.ab.android.sdk.EyeofcloudClient;
import com.eyeofcloud.ab.android.sdk.EyeofcloudStartListener;

public class LoadingActivity extends Activity {

    private MyApplication application;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_loading);
        application = (MyApplication) getApplication();
    }

    @Override
    protected void onStart() {
        super.onStart();

        application.getEyeofcloudManager().initialize(this, new EyeofcloudStartListener() {
            @Override
            public void onStart(EyeofcloudClient eyeofcloud) {
                //sleep for showing loading UI
                try {
                    Thread.sleep(1000);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                Intent intent = new Intent(application.getBaseContext(), HomeActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: Demo/AndroidFullStackKotlinSimple/AndroidFullStackActivateIntent/app/src/main/java/com/example/eyeofcloud/androidfullstackactivateintent/LoadingActivity.kt
package com.example.eyeofcloud.androidfullstackactivateintent

import android.app.Activity
import android.content.Intent
import android.os.Bundle



class LoadingActivity : Activity() {

    private var application: MyApplication? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_loading)
        application = getApplication() as MyApplication
    }

    override fun onStart() {
        super.onStart()

        application!!.eyeofcloudManager!!.initialize(this) {
            //sleep for showing loading UI
            try {
                Thread.sleep(1000)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

            val intent = Intent(application!!.baseContext, HomeActivity::class.java)
            startActivity(intent)
        }
    }
}
